nebula_ros/src/continental/continental_ars548_hw_interface_ros_wrapper.cpp

```diff
@@ -0,0 +1,329 @@
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("data_port", 2368, descriptor);
+    sensor_configuration.data_port = this->get_parameter("data_port").as_int();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("configuration_host_port", 2368, descriptor);
+    sensor_configuration.configuration_host_port =
+      this->get_parameter("configuration_host_port").as_int();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<uint16_t>("configuration_sensor_port", 2368, descriptor);
+    sensor_configuration.configuration_sensor_port =
+      this->get_parameter("configuration_sensor_port").as_int();
+  }
+
+  if (sensor_configuration.sensor_model == nebula::drivers::SensorModel::UNKNOWN) {
+    return Status::INVALID_SENSOR_MODEL;
+  }
+
+  if (sensor_configuration.frame_id.empty()) {
+    return Status::SENSOR_CONFIG_ERROR;
+  }
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "SensorConfig:" << sensor_configuration);
+  return Status::OK;
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::ReceivePacketsDataCallback(
+  std::unique_ptr<nebula_msgs::msg::NebulaPackets> scan_buffer)
+{
+  packets_pub_->publish(std::move(scan_buffer));
+}
+
+rcl_interfaces::msg::SetParametersResult ContinentalARS548HwInterfaceRosWrapper::paramCallback(
+  const std::vector<rclcpp::Parameter> & p)
+{
+  std::scoped_lock lock(mtx_config_);
+  RCLCPP_DEBUG_STREAM(this->get_logger(), "add_on_set_parameters_callback");
+  RCLCPP_DEBUG_STREAM(this->get_logger(), p);
+  RCLCPP_DEBUG_STREAM(this->get_logger(), sensor_configuration_);
+  RCLCPP_INFO_STREAM(this->get_logger(), p);
+
+  drivers::ContinentalARS548SensorConfiguration new_param{sensor_configuration_};
+  RCLCPP_INFO_STREAM(this->get_logger(), new_param);
+  std::string sensor_model_str;
+  std::string return_mode_str;
+  if (
+    get_param(p, "sensor_model", sensor_model_str) ||
+    get_param(p, "return_mode", return_mode_str) || get_param(p, "host_ip", new_param.host_ip) ||
+    get_param(p, "sensor_ip", new_param.sensor_ip) ||
+    get_param(p, "frame_id", new_param.frame_id) ||
+    get_param(p, "data_port", new_param.data_port) ||
+    get_param(p, "configuration_host_port", new_param.configuration_host_port) ||
+    get_param(p, "configuration_sensor_port", new_param.configuration_sensor_port)) {
+    if (0 < sensor_model_str.length())
```

Comment:
I think these should be `|` instead of `||`. Because `||` is a short-circuit operator, all parameters in `p` after the 1st one are ignored.

Reply:
:scream: 
This a recurrent bug in nebula then, as all wrappers to the same. Would you prefer a local fix here and a separate PR for all the others? Or can I fix them all here?

Reply:
@mojomex 
Thinking on how we did not find it so far, it is because when you modify parameters via UI, you end up modifying one by one. Of course, it is still dangerous so your approach is still needed

Reply:
@mojomex 
It seems this bug actually gave David headaches lately :rofl:

Reply:
Linked issue: 
* #108 

I think we should fix the Continental one directly in this PR and we'll keep track of the other vendors in the above issue!

Reply:
Thanks !
Addressed in [2338af7](https://github.com/tier4/nebula/pull/107/commits/2338af727c305890391d6daa356c55319e204dc5)